### PR TITLE
Refactor ArtifactFileField

### PR DESCRIPTION
**Pull request by goosemania**
closes #4382
https://pulp.plan.io/issues/4382

### CODE REVIEW

**Comment by goosemania** [comment]
@daviddavis , if you can try it on S3 that would be great. Thank you!

File: pulpcore/app/models/fields.py [hunk 1]
```diff
@@ -9,37 +9,49 @@
...
+        if file.name != artifact_storage_path and file.name.startswith(
                 os.path.join(settings.MEDIA_ROOT, 'artifact')):
             raise ValueError(_('The file referenced by the Artifact is already present in '
                                'Artifact storage. Files must be stored outside this location '
                                'prior to Artifact creation.'))
-        file = super().pre_save(model_instance, add)
-        if file and file._committed and add and str(file) != upload_to:
-            # the file needs to be moved into place
+
+        move = (file._committed and file.name != artifact_storage_path)
```
Comment (daviddavis) [inline pulpcore/app/models/fields.py]: Why define a `move` variable here?

**Comment by daviddavis** [comment]
This looks like a huge improvement! Will test it out in S3.

File: pulpcore/app/models/fields.py (hunk 1 above)
Comment (goosemania) [reply pulpcore/app/models/fields.py]: I tried to make it more readable so it's clear what this conditions are deciding. 
I can drop it.

Comment (daviddavis) [reply pulpcore/app/models/fields.py]: Either way works for me. I'll leave it up to you.

**Comment by bmbouter** [comment]
This is a big improvement. Thanks @goosemania ! Thank you for testing on S3 @daviddavis ! :+1:

**Pull request merged**